src/timezones/local.jl

```diff
@@ -0,0 +1,237 @@
+end
+
+function _get_localzone_windows()
+    win_tz_path = joinpath(WINDOWS_DIR, "windowszones_compiled")
+
+    isfile([win_tz_path]...) ||
+        error("Failed: Windows zones not found. Try running Pkg.build(\"TimeZones\")")
+
+    win_tz_dict = nothing
+    open(win_tz_path, "r") do fp
+        win_tz_dict = deserialize(fp)
+    end
+
+    winzone = strip(readall(`powershell -Command "[TimeZoneInfo]::Local.Id"`))
+    if haskey(win_tz_dict, winzone)
+        timezone = win_tz_dict[winzone]
+    else
+        error("Failed to determine your Windows timezone. ",
+                "Uses powershell, should work on windows 7 and above")
+    end
+
+    return TimeZone(timezone)
+end
+
+type TTInfo
+    tt_gmtoff::Int32
+    tt_isdst::Int8
+    tt_abbrind::Uint8
+end
+
+function build_tzinfo(zone::String, filename::String)
+    file = open([filename]...)
+    try
+        # Reference: http://man7.org/linux/man-pages/man5/tzfile.5.html
+        magic = readbytes(file, 4)
+        @assert magic == b"TZif" "Magic file identifier \"TZif\" not found."
+        version = readbytes(file, 1)
+        # Fifteen bytes containing zeros reserved for future use.
+        readbytes(file, 15)
+        # The number of UTC/local indicators stored in the file.
+        tzh_ttisgmtcnt = bswap(read(file, Int32))
+        # The number of standard/wall indicators stored in the file.
+        tzh_ttisstdcnt = bswap(read(file, Int32))
+        # The number of leap seconds for which data is stored in the file.
+        tzh_leapcnt = bswap(read(file, Int32))
+        # The number of "transition times" for which data is stored in the file.
+        tzh_timecnt = bswap(read(file, Int32))
+        # The number of "local time types" for which data is stored in the file (must not be zero).
+        tzh_typecnt = bswap(read(file, Int32))
+        # The number of characters of "timezone abbreviation strings" stored in the file.
+        tzh_charcnt = bswap(read(file, Int32))
+
+        transitions = Array{Int32}(tzh_timecnt)
+        for index in 1:tzh_timecnt
+            transitions[index] = bswap(read(file, Int32))
+        end
+        lindexes = Array{UInt8}(tzh_timecnt)
+        for index in 1:tzh_timecnt
+            lindexes[index] = bswap(read(file, UInt8)) + 1 # Julia uses 1 indexing
+        end
+        ttinfo = Array{TTInfo}(tzh_typecnt)
+        for index in 1:tzh_typecnt
+            ttinfo[index] = TTInfo(
+                    bswap(read(file, Int32)),
+                    bswap(read(file, Int8)),
+                    bswap(read(file, Uint8)) + 1 # Julia uses 1 indexing
+                )
+        end
+        tznames_raw = Array{UInt8}(tzh_charcnt)
+        namestart = 1
+        tznames = Dict{Uint8, String}()
+        for index in 1:tzh_charcnt
+            tznames_raw[index] = bswap(read(file, UInt8))
+            if tznames_raw[index] == '\0'
+                tznames[namestart] = ascii(tznames_raw[namestart:index-1])
```

Comment:
If `index = 1` then you could have an exception being raised from `ascii([])`. Is this a problem?